File: car-rental-front/src/app/modules/customer/service/customer.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { StorageService } from '../../../auth/services/storage/storage.service';

@Injectable({
  providedIn: 'root'
})
export class CustomerService {

   BaseUrl = "http://localhost:9000/api/customer";

  constructor(private httpClient: HttpClient) { }

  getAllCars(): Observable<any> {
    return this.httpClient.get(this.BaseUrl + `/cars`, {
      withCredentials: true
    })
  }

  getCarsPage(page: number, size: number): Observable<any> {
      return this.httpClient.get(this.BaseUrl + `/cars-page?page=${page-1}&size=${size}`, {
        withCredentials: true
      })
  }

  getCarById(carId: number): Observable<any> {
    return this.httpClient.get(this.BaseUrl + `/car/${carId}`, {
      withCredentials: true,
    })
  }

  bookCar(bookCarDTO: any): Observable<any> {
    return this.httpClient.post(this.BaseUrl + `/book-car`, bookCarDTO, {
      withCredentials: true,
    })
  }

  getPageOfBookingHistory(page: number, size: number): Observable<any> {
    const userId = StorageService.getUserId();
    return this.httpClient.get(this.BaseUrl + `/booking-history-page/${userId}?page=${page-1}&size=${size}`, {
      withCredentials: true
    })
  }
}

File: car-rental-spring-modify/src/main/java/com/coding/car_rental_app/services/customer/CustomerService.java
package com.coding.car_rental_app.services.customer;

import com.coding.car_rental_app.dtos.BookingDetailsDTO;
import com.coding.car_rental_app.dtos.CarDTO;
import com.coding.car_rental_app.dtos.CarDTOPage;
import com.coding.car_rental_app.entity.Car;
import org.springframework.data.domain.Page;

import java.util.List;

public interface CustomerService {
    List<CarDTO> findAllCars();

    CarDTOPage getAllCars(int page, int size);

    CarDTO getCarById(Long carId);

    boolean bookCar(BookingDetailsDTO bookingDetailsDTO);
}
